jans-cedarling/cedarling/src/common/cedar_schema.rs

```diff
@@ -6,31 +6,153 @@
  */
 
 pub(crate) use cedar_json::CedarSchemaJson;
-
 pub(crate) mod cedar_json;
 
+/// cedar_schema value which specifies both encoding and content_type
+///
+/// encoding is one of none or base64
+/// content_type is one of cedar or cedar-json#[derive(Debug, Clone, serde::Deserialize)]
+#[derive(Debug, Clone, serde::Deserialize)]
+struct EncodedSchema {
+    pub encoding : super::Encoding,
+    pub content_type : super::ContentType,
+    pub body : String,
+}
+
+/// Intermediate struct to handle both kinds of cedar_schema values.
+///
+/// Either
+///   "cedar_schema": "cGVybWl0KA..."
+/// OR
+///   "cedar_schema": { "encoding": "...", "content_type": "...", "body": "permit(...)"}#[derive(Debug, Clone, serde::Deserialize)]
+#[derive(Debug, Clone, serde::Deserialize)]
+#[serde(untagged)]
+enum MaybeEncoded {
+    Plain(String),
+    Tagged(EncodedSchema)
+}
+
 /// Box that holds the [`cedar_policy::Schema`] and
 /// JSON representation that is used to create entities from the schema in the policy store.
 #[derive(Debug, Clone)]
-#[allow(dead_code)]
 pub(crate) struct CedarSchema {
     pub schema: cedar_policy::Schema,
     pub json: cedar_json::CedarSchemaJson,
 }
 
+impl PartialEq for CedarSchema {
+    fn eq(&self, other: &Self) -> bool {
+        // Have to check principals, resources, action_groups, entity_types,
+        // actions. Those can contain duplicates, and are not stored in comparison order.
+        // So use HashSet to compare them.
+
+        let self_principals = self.schema.principals().collect::<std::collections::HashSet<_>>();
```

Comment:
Why you not import HashSet?
Do you like long form usage `std::collections::HashSet`?

Reply:
I try to keep `use` as local as possible. Which I know is a little uncommon.

But thanks - this one was a cut'n'paste and should be at least local to the function.